### PR TITLE
Linedensity

### LINKED ISSUE #4763: Request in QGIS ([feature][processing] Add Line density algorithm)
### Request for documentation
From pull request QGIS/qgis#33525
Author: @root676
QGIS version: 

**[feature][processing] Add Line density algorithm**

## Description
This PR proposes a _Line density algorithm_ as it is implemented in [ESRI ArcGIS Spatial Analyst](https://desktop.arcgis.com/en/arcmap/latest/tools/spatial-analyst-toolbox/how-line-density-works.htm). The algorithm basically moves a circular search geometry from cell to cell and calculates the line density measure as proposed by Silverman (1986). The measure is a weighted sum of intersecting line lengths that are intersecting with the search geometry, which is then divided by the search geometry area. The figure below gives a rough example of the algorithms functionality.

LineDensity_alg

The algorithm output for the road network of Vienna ([Data Source: Stadt Wien - data.wien.gv.at](https://www.data.gv.at/katalog/dataset/1039ed7e-97fb-435f-b6cc-f6a105ba5e09)) with randomly distributed weights between 1 and 3 looks like this:

LineDensityExample

Please feel free to review/comment/suggest improvements!

Literature:

- Silverman, B. W. Density Estimation for Statistics and Data Analysis. New York: Chapman and Hall, 1986.

## Checklist

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] Commits which fix bugs include `Fixes #11111` at the bottom of the commit message
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md#contributing-to-qgis) before each commit
- [x] I have evaluated whether it is appropriate for this PR to be backported, backport requests are left as label or comment

### Commits tagged with [need-docs]
"[feature][processing] Add Line denisty algorithm"

### PR DESCRIPTION
**Pull request by rosaguilar**
Documenting line density alg.
Goal: fix #4763

### CODE REVIEW

File: docs/user_manual/processing_algs/qgis/interpolation.rst [hunk 1]
```diff
@@ -352,6 +352,97 @@ Outputs
...
+     - Radius of the circular neighbourhood. Units can be specified here.
+
+       Default: 10
+
+   * - **Pixel size**
+     - ``PIXEL_SIZE``
+     - [number]
+     - Pixel size of the output raster layer in layer units.
+       The raster has square pixels.
+       Default: 10
```
Comment (havatv) [inline docs/user_manual/processing_algs/qgis/interpolation.rst:406]:
```suggestion
     - [number]

       Default: 10
     - Pixel size of the output raster layer in layer units.
       The raster has square pixels.
```

File: docs/user_manual/processing_algs/qgis/interpolation.rst [hunk 2]
```diff
@@ -352,6 +352,97 @@ Outputs
...
+     - ``WEIGHT``
+     - [number]
+     - Field of the layer containing the weight factor to use during the calculation
+
+   * - **Search Radius**
+     - ``RADIUS``
+     - [number]
+     - Radius of the circular neighbourhood. Units can be specified here.
+
+       Default: 10
```
Comment (havatv) [inline docs/user_manual/processing_algs/qgis/interpolation.rst:399]:
```suggestion
     - [number]

       Default: 10
     - Radius of the circular neighbourhood. Units can be specified here.
```

File: docs/user_manual/processing_algs/qgis/interpolation.rst [hunk 3]
```diff
@@ -352,6 +352,97 @@ Outputs
...
+     - ``OUTPUT``
+     - [raster]
+
+       Default: ``[Save to temporary file]``
+     - The output as a raster layer. One of:
+
+       * Save to a Temporary File
+       * Save to File...
+
+       The file encoding can also be changed here.
```
Comment (havatv) [inline docs/user_manual/processing_algs/qgis/interpolation.rst:443]:
```suggestion
     - The output line density raster layer.
```

File: docs/user_manual/processing_algs/qgis/interpolation.rst [hunk 4]
```diff
@@ -352,6 +352,97 @@ Outputs
...
+.. list-table::
+   :header-rows: 1
+   :widths: 20 20 20 40
+   :stub-columns: 0
+
+   * - Label
+     - Name
+     - Type
+     - Description
+
```
Comment (havatv) [inline docs/user_manual/processing_algs/qgis/interpolation.rst:383]:
```suggestion
```

File: docs/user_manual/processing_algs/qgis/interpolation.rst [hunk 5]
```diff
@@ -352,6 +352,97 @@ Outputs
...
+   * - **Input line layer**
+     - ``INPUT``
+     - [vector: any]
+     - Input vector layer containing line features
+
```
Comment (havatv) [inline docs/user_manual/processing_algs/qgis/interpolation.rst:388]:
```suggestion
```

File: docs/user_manual/processing_algs/qgis/interpolation.rst [hunk 6]
```diff
@@ -352,6 +352,97 @@ Outputs
...
+   * - **Weight field**
+     - ``WEIGHT``
+     - [number]
+     - Field of the layer containing the weight factor to use during the calculation
+
```
Comment (havatv) [inline docs/user_manual/processing_algs/qgis/interpolation.rst:393]:
```suggestion
```

File: docs/user_manual/processing_algs/qgis/interpolation.rst [hunk 7]
```diff
@@ -352,6 +352,97 @@ Outputs
...
+   * - **Search Radius**
+     - ``RADIUS``
+     - [number]
+     - Radius of the circular neighbourhood. Units can be specified here.
+
+       Default: 10
+
```
Comment (havatv) [inline docs/user_manual/processing_algs/qgis/interpolation.rst:400]:
```suggestion
```

File: docs/user_manual/processing_algs/qgis/interpolation.rst [hunk 8]
```diff
@@ -352,6 +352,97 @@ Outputs
...
+   * - **Pixel size**
+     - ``PIXEL_SIZE``
+     - [number]
+     - Pixel size of the output raster layer in layer units.
+       The raster has square pixels.
+       Default: 10
+
```
Comment (havatv) [inline docs/user_manual/processing_algs/qgis/interpolation.rst:407]:
```suggestion
```

**Comment by havatv** [comment]
Thanks, @rosaguilar!

**Comment by rosaguilar** [comment]
> Thanks, @rosaguilar!

to you as well. :)

**Comment by havatv** [comment]
@rosaguilar, in which QGIS version was this algorithm added?
If 3.10 (current LTR) or earlier, this should be backported.

**Comment by rosaguilar** [comment]
> @rosaguilar, in which QGIS version was this algorithm added?
> If 3.10 (current LTR) or earlier, this should be backported.
@havatv, 3.12: 
https://www.qgis.org/en/site/forusers/visualchangelog312/index.html#feature-line-density-algorithm

**Comment by root676** [comment]
@rosaguilar Thank you!!

**Pull request merged**